packages/http-crawler/src/internals/http-crawler.ts

```diff
@@ -0,0 +1,845 @@
+            preNavigationHooks = [],
+            postNavigationHooks = [],
+
+            // Ignored
+            handleRequestFunction,
+
+            // BasicCrawler
+            autoscaledPoolOptions = HTTP_OPTIMIZED_AUTOSCALED_POOL_OPTIONS,
+            ...basicCrawlerOptions
+        } = options;
+
+        super({
+            ...basicCrawlerOptions,
+            requestHandler,
+            autoscaledPoolOptions,
+            // We need to add some time for internal functions to finish,
+            // but not too much so that we would stall the crawler.
+            requestHandlerTimeoutSecs: navigationTimeoutSecs + requestHandlerTimeoutSecs + BASIC_CRAWLER_TIMEOUT_BUFFER_SECS,
+        }, config);
+
+        this._handlePropertyNameChange({
+            newName: 'requestHandler',
+            oldName: 'handlePageFunction',
+            propertyKey: 'requestHandler',
+            newProperty: requestHandler,
+            oldProperty: handlePageFunction,
+            allowUndefined: true,
+        });
+
+        if (!this.requestHandler) {
+            this.requestHandler = this.router;
+        }
+
+        // Cookies should be persisted per session only if session pool is used
+        if (!this.useSessionPool && persistCookiesPerSession) {
+            throw new Error('You cannot use "persistCookiesPerSession" without "useSessionPool" set to true.');
+        }
+
+        this.supportedMimeTypes = new Set([...HTML_AND_XML_MIME_TYPES, APPLICATION_JSON_MIME_TYPE]);
+        if (additionalMimeTypes.length) this._extendSupportedMimeTypes(additionalMimeTypes);
+
+        if (suggestResponseEncoding && forceResponseEncoding) {
+            this.log.warning('Both forceResponseEncoding and suggestResponseEncoding options are set. Using forceResponseEncoding.');
+        }
+
+        this.userRequestHandlerTimeoutMillis = requestHandlerTimeoutSecs * 1000;
+        this.navigationTimeoutMillis = navigationTimeoutSecs * 1000;
+        this.ignoreSslErrors = ignoreSslErrors;
+        this.suggestResponseEncoding = suggestResponseEncoding;
+        this.forceResponseEncoding = forceResponseEncoding;
+        this.proxyConfiguration = proxyConfiguration;
+        this.preNavigationHooks = preNavigationHooks;
+        this.postNavigationHooks = [
+            ({ request, response }) => this._abortDownloadOfBody(request, response!),
+            ...postNavigationHooks,
+        ];
+
+        if (this.useSessionPool) {
+            this.persistCookiesPerSession = persistCookiesPerSession ?? true;
+        } else {
+            this.persistCookiesPerSession = false;
+        }
+    }
+
+    /**
+     * **EXPERIMENTAL**
+     * Function for attaching CrawlerExtensions such as the Unblockers.
+     * @param extension Crawler extension that overrides the crawler configuration.
+     */
+    use(extension: CrawlerExtension) {
+        ow(extension, ow.object.instanceOf(CrawlerExtension));
+
+        const clazz = this.constructor.name;
```

Comment:
i dont mind this, but `className` is generally better than coming up with non existing spelling :] or just `cls`, but that's just me being into shortcuts :D

Reply:
`className` sounds good